SonarQube.Client/Models/ServerSentEvents/IServerSentEventsSession.cs

```diff
@@ -0,0 +1,30 @@
+﻿/*
+ * SonarQube Client
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Threading.Tasks;
+using SonarQube.Client.Models.ServerSentEvents.ClientContract;
+
+namespace SonarQube.Client.Models.ServerSentEvents
+{
+    public interface IServerSentEventsSession
+    {
+        Task<IServerEvent> ReadAsync();
```

Comment:
Yes, it should be cancellable. We were planning to store the cancellation token that is provided to the `CreateServerSentEventsSession` method, and it would get cancelled on solution change and VS shutdown

Reply:
Well, not sure we necessarily need to on VS shutdown, but on solution change for sure, to reinitialize the events session for the new project